aptos-move/framework/aptos-framework/sources/TransactionFee.move

```diff
@@ -1,10 +1,22 @@
 module AptosFramework::TransactionFee {
-    use AptosFramework::TestCoin::{Self, Coin};
+    use AptosFramework::Coin::{Self, BurnCapability, Coin};
+    use AptosFramework::TestCoin::TestCoin;
 
     friend AptosFramework::Account;
 
+    struct TestCoinCapabilities has key {
+        burn_cap: BurnCapability<TestCoin>,
+    }
+
     /// Burn transaction fees in epilogue.
-    public(friend) fun burn_fee(fee: Coin) {
-        TestCoin::burn_gas(fee);
+    public(friend) fun burn_fee(fee: Coin<TestCoin>) acquires TestCoinCapabilities {
+        Coin::burn_with_capability<TestCoin>(
+            fee,
+            &borrow_global<TestCoinCapabilities>(@AptosFramework).burn_cap,
```

Comment:
i don't think we need to store a wrapper here, we can have a friend function burn in TestCoin

Reply:
I discussed with David yesterday and it's a bit cleaner to not have friends at all. This allows all mint/burn to go through the same standardized Coin mint/burn flows.

Reply:
honestly, we should just put this into account.